src/core/annotation.js

```diff
@@ -1606,12 +1681,26 @@ class WidgetAnnotation extends Annotation {
     dict.set("AP", AP);
     dict.set("M", `D:${getModificationDate()}`);
 
+    if (rotation || this.borderColor || this.backgroundColor) {
+      const MK = new Dict(xref);
+      MK.set("R", rotation);
+      MK.set("BC", getPDFRgbArray(this.borderColor));
+      MK.set("BG", getPDFRgbArray(this.backgroundColor));
+      dict.set("MK", MK);
+    }
```

Comment:
Maybe if you add a method like this, on the `WidgetAnnotation`-class:
```javascript
_getMKDict(rotation) {
  const mk = new Dict(null); // `null` can probably be used here, since none of the data below will be a /Ref
  if (rotation) {
    mk.set("R", rotation);
  }
  if (this.borderColor) {
    mk.set("BC", Array.from(this.borderColor).map(c => c / 255));
  }
  if (this.backgroundColor) {
    mk.set("BG", Array.from(this.backgroundColor).map(c => c / 255));
  }
  return mk.size > 0 ? mk : null;
}
```

and then you can simply do (here and elsewhere):
```suggestion
    const maybeMK = this._getMKDict(rotation);
    if (maybeMK) {
      dict.set("MK", maybeMK);
    }
```